tools/ui-components/src/provide-context/index.ts

```diff
@@ -0,0 +1,7 @@
+import { createContext } from 'react';
+
+export type ContextProps<T = string> = {
+  data?: T;
+};
```

Comment:
This is still simple, because we don't have alert of something alike that need context yet, when cross that bridge we shall worry about it.